library/src/main/java/com/chuckerteam/chucker/internal/support/OkioUtils.kt

```diff
@@ -0,0 +1,24 @@
+package com.chuckerteam.chucker.internal.support
+
+import okio.Buffer
+import java.io.EOFException
+
+private const val MAX_PREFIX_SIZE = 64L
+private const val CODE_POINT_SIZE = 16
+
+/**
+ * Returns true if the body in question probably contains human readable text. Uses a small sample
+ * of code points to detect unicode control characters commonly used in binary file signatures.
+ */
+internal val Buffer.isProbablyPlainText
+    get() = try {
+        val prefix = Buffer()
+        val byteCount = if (size < MAX_PREFIX_SIZE) size else MAX_PREFIX_SIZE
+        copyTo(prefix, 0, byteCount)
+        sequence { while (!prefix.exhausted()) yield(prefix.readUtf8CodePoint()) }
+            .take(CODE_POINT_SIZE)
+            .any { codePoint -> Character.isISOControl(codePoint) && !Character.isWhitespace(codePoint) }
+            .not()
```

Comment:
I have to admit that the logic here is a bit hard to follow due to `!` and `.not`. Could we reword it to a single expression using `.all{}` or `.none{}`?

Reply:
Simplified :).